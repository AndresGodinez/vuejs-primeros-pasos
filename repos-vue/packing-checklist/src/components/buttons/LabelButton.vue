<template>
  <button class="btn-label" @click="onClick">
    <img :src="image" alt="Icon" height="14" />
    <slot>Button</slot>
  </button>
</template>

<script>
export default {
  name: 'label-button',
  props: {
    image: {
      type: String,
      required: true
    },
    onClick: {
      type: Function,
      required: true
    }
  }
}
</script>

<style lang="sass" scoped>
@import '~@/sass/_vars.sass'
@import '~@/sass/_buttons.sass'

.btn-label
  @extend %button-base
  border: 1px solid $main-color
  padding: 1em
</style>
