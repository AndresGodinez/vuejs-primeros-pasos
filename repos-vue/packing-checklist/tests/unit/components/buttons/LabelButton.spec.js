import { shallow } from '@vue/test-utils'
import LabelButton from '@/components/buttons/LabelButton.vue'

describe('LabelButton', () => {
  it('has the right DOM structure', () => {
    const wrapper = shallow(LabelButton, {
      propsData: { image: 'algo.svg', onClick: function () {} }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
