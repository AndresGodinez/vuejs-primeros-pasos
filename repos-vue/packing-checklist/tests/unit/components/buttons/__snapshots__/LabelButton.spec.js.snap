// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelButton has the right DOM structure 1`] = `
<button
  class="btn-label"
>
  <img
    alt="Icon"
    height="14"
    src="algo.svg"
  />
   Button
</button>
`;
